### PR TITLE
Updated C28719

### PR DESCRIPTION
**Pull request by MugBergerFries**
Added an example and a list of all banned functions covered by this rule

### CODE REVIEW

File: windows-driver-docs-pr/devtest/28719-banned-api-usage-use-updated-function-replacement.md [hunk 1]
```diff
@@ -9,11 +9,146 @@ f1_keywords:
 # C28719
 
 
-warning C28719: Banned API Usage
+**warning C28719: Banned API Usage**\
```
Comment (MichaelSquires) [inline windows-driver-docs-pr/devtest/28719-banned-api-usage-use-updated-function-replacement.md:12]:
```suggestion
**Warning C28719: Banned API Usage**\
```

File: windows-driver-docs-pr/devtest/28719-banned-api-usage-use-updated-function-replacement.md [hunk 2]
```diff
@@ -9,11 +9,146 @@ f1_keywords:
...
+_Banned API Usage:  *function name* is a Banned API as listed in dontuse.h for security purposes._
```
Comment (MichaelSquires) [inline windows-driver-docs-pr/devtest/28719-banned-api-usage-use-updated-function-replacement.md:13]: Is `dontuse.h` a public header? If not, we should remove this reference.

File: windows-driver-docs-pr/devtest/28719-banned-api-usage-use-updated-function-replacement.md [hunk 3]
```diff
@@ -9,11 +9,146 @@ f1_keywords:
...
 
-This warning indicates that a function is being used that has been banned, and has a more robust and secure replacement.
+This warning indicates that a function is being used that has been banned and has a more robust or secure replacement.
+A list of all banned functions covered by this error, why they are banned, and recommended replacements can be found after the following example: 
 
- 
+## Example 
+
+The following code generates this warning: 
+```
```
Comment (MichaelSquires) [inline windows-driver-docs-pr/devtest/28719-banned-api-usage-use-updated-function-replacement.md:21]: See if you can get syntax highlighting working for this. I posted a suggestion in another PR.

File: windows-driver-docs-pr/devtest/28719-banned-api-usage-use-updated-function-replacement.md [hunk 4]
```diff
@@ -9,11 +9,146 @@ f1_keywords:
...
+    char dst[100]; 
+    strcpy(dst, src);
+} 
+```
+This is due to the use of the unsafe function strcpy.
+strcpy does not check if the destination buffer is large enough to fit the source data.
+To fix this issue, we can use strcpy_s, C++11’s safe replacement to this function.
+strcpy_s has a third parameter (the size of the destination buffer) to ensure only that many bytes are copied.
+For example, the following code is safe: 
+```
```
Comment (MichaelSquires) [inline windows-driver-docs-pr/devtest/28719-banned-api-usage-use-updated-function-replacement.md:33]: Syntax highlighting?

File: windows-driver-docs-pr/devtest/28719-banned-api-usage-use-updated-function-replacement.md [hunk 5]
```diff
@@ -9,11 +9,146 @@ f1_keywords:
...
+void example_func(PSTR src) 
+{ 
+    char dst[100]; 
+    strcpy_s(dst, sizeof(dst), src); 
+}
+```
+## Banned Functions  
+_NOTE: This list is actively being updated and improved_
+ | banned API | replacement(s) | rationale / notes |
```
Comment (MichaelSquires) [inline windows-driver-docs-pr/devtest/28719-banned-api-usage-use-updated-function-replacement.md:42]:
```suggestion
 | Banned API | Replacement Function(s) | Rationale / Notes |
```

**Review by MichaelSquires — CHANGES_REQUESTED** [review]

File: windows-driver-docs-pr/devtest/28719-banned-api-usage-use-updated-function-replacement.md (hunk 3 above)
Comment (MugBergerFries) [reply windows-driver-docs-pr/devtest/28719-banned-api-usage-use-updated-function-replacement.md]: Added!

File: windows-driver-docs-pr/devtest/28719-banned-api-usage-use-updated-function-replacement.md (hunk 4 above)
Comment (MugBergerFries) [reply windows-driver-docs-pr/devtest/28719-banned-api-usage-use-updated-function-replacement.md]: Added!

**Review by MichaelSquires — APPROVED** [review]

File: windows-driver-docs-pr/devtest/28719-banned-api-usage-use-updated-function-replacement.md [hunk 6]
```diff
@@ -9,11 +9,146 @@ f1_keywords:
 # C28719
 
 
-warning C28719: Banned API Usage
+**Warning C28719: Banned API Usage**\
+Example output: ```Banned API Usage:  *function name* is a Banned API as listed in dontuse.h for security purposes.```
```
[the remark this answers is not in the document]
Comment (MugBergerFries) [reply windows-driver-docs-pr/devtest/28719-banned-api-usage-use-updated-function-replacement.md]: Changed to reflect

**Comment by MugBergerFries** [comment]
@DOMARS This is approved and ready to be merged

**Comment by DOMARS** [comment]
@MugBergerFries - Will merge this to live.

**Pull request merged**